Kryptonite/Silo.swift

```diff
@@ -274,31 +282,33 @@ class Silo {
                     case .tag(let tag):
                         LogManager.shared.save(theLog: TagSignatureLog(session: session.id, signature: TagSignatureLog.rejectedConstant, tag: tag), deviceName: session.pairing.name)
                     }
-
+                    
                     throw UserRejectedError()
                 }
                 
             }  catch {
                 err = "\(error)"
             }
             
-            gitSign = GitSignResponse(sig: sig, err: err)
-        }
-        
-        if let meRequest = request.me {
+            responseType = .git(GitSignResponse(sig: sig, err: err))
+            
+        case .me(let meRequest):
             let keyManager = try KeyManager.sharedInstance()
             
             var pgpPublicKey:Data?
             if let pgpUserID = meRequest.pgpUserId {
                 pgpPublicKey = try keyManager.loadPGPPublicKey(for: pgpUserID).packetData
             }
             
-            me = MeResponse(me: MeResponse.Me(email: try keyManager.getMe(), publicKeyWire: try keyManager.keyPair.publicKey.wireFormat(), pgpPublicKey: pgpPublicKey))
+            responseType = .me(MeResponse(me: MeResponse.Me(email: try keyManager.getMe(), publicKeyWire: try keyManager.keyPair.publicKey.wireFormat(), pgpPublicKey: pgpPublicKey)))
+
+        default:
+            throw ResponseNotNeededError()
```

Comment:
it's not expected behavior as responseFor cannot create responses for noOp and unpair request bodies.